x/concentrated-liquidity/swaps.go

```diff
@@ -142,6 +142,13 @@ func (k Keeper) SwapExactAmountIn(
 		return osmomath.Int{}, err
 	}
 
+	// Trigger before hook for SwapExactAmountIn prior to mutating state.
+	// If no contract is set, this will be a no-op.
+	err = k.BeforeSwapExactAmountIn(ctx, pool.GetId(), sender, tokenIn, tokenOutDenom, tokenOutMinAmount, spreadFactor)
+	if err != nil {
+		return osmomath.Int{}, err
+	}
```

Comment:
I don't think we have a mechanism in our tests for validating that `BeforeSwapExactAmountIn` is triggered **before** and not after the swap.

I think it would be valuable to add such an assertion so that we minimize the chances of a move-by-mistake in the future

Reply:
This seems like it will require a meaningful change to the current testing approach so I created a separate issue with what I think would be a viable design in #6862 

I agree that it's valuable to add this to ensure future changes/additions to hooks don't break this invariant. Would be great if we can handle this separately as I think it will take careful implementation to keep tests readable and would like to get the remaining core feature work in first